lib/Database/lmdb/mdb.c

```diff
@@ -144,7 +144,7 @@ typedef SSIZE_T	ssize_t;
 #include <unistd.h>
 #endif
 
-#if defined(__sun) || defined(ANDROID)
+#if defined(__sun)
```

Comment:
`posix_memalign` was [added to Bionic with 4.2, ie API 17](https://android.googlesource.com/platform/bionic/+/master/libc/include/stdlib.h#80), a long time ago, which is just one higher than the current minimum supported API of 16 in the NDK, so I think it's safe to assume it's there.

Reply:
This change should be upstreamed.  I see the upstream fixed the `ANDROID` vs. `__ANDROID__` issue, but this code is still disable on Android: https://github.com/LMDB/lmdb/blob/c3e6b4209eed13af4a3670e5f04f42169c08e5c6/libraries/liblmdb/mdb.c#L147

Is this change *necessary* to get Android working, or just an incremental improvement?  If it's necessary, we could take this change immediately.  If not, my preference would be to get this change in upstream and then update our version of lmdb.  CC @akyrtzi in case he feels differently.

Reply:
No, it's not necessary, I just went ahead and updated it, removed now.